pkg/cosign/tuf/client.go

```diff
@@ -186,7 +192,11 @@ func getRoot(meta map[string]json.RawMessage) (json.RawMessage, error) {
 		return trustedRoot, nil
 	}
 	// On first initialize, there will be no root in the TUF DB, so read from embedded.
-	trustedRoot, err := embeddedRootRepo.ReadFile(path.Join("repository", "root.json"))
+	rd, ok := GetEmbedded().(fs.ReadFileFS)
+	if !ok {
+		return nil, errors.New("fs.ReadFileFS unimplemented for embedded repo")
```

Comment:
This could just be that I'm bad at Go.

Reply:
I'm pretty bad at it too! It's because `GetEmbedded` returns a `fs.FS`; there wasn't another interface that could both support `embed.FS` (for the actual prod embedded files) and a testable filesystems.

https://cs.opensource.google/go/go/+/go1.18.2:src/embed/embed.go;l=148

(embed.FS implements fs.FS)

It *shouldn't* happen in practice since prod clients are using `embed.FS` but I needed to be able to swap the embedded out for test. I could panic here, but preferred to return the error.

Reply:
Ahh okay, sounds like it's a limitation of the type system. Feel free to resolve